### ISSUE TITLE
Chaining decorators

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#2851]
### Discussed in https://github.com/microsoft/pyright/discussions/2850

<div type='discussions-op-text'>

<sup>Originally posted by **lovetox** January 11, 2022</sup>
Hi,

I seem to have problems with chanining decorators

```python
from __future__ import annotations

from typing import Callable

import functools


def decorator(func: Callable[[int, str], None]) -> Callable[[int, str], None]:
    @functools.wraps(func)
    def func_wrapper(firstarg: int, secondarg: str) -> None:
        return func(firstarg, secondarg)
    return func_wrapper


class Test:
    def __init__(self):
        self.test(1, 'a')

    @staticmethod
    @decorator
    def test(firstarg: int, secondarg: str) -> None:
        print(firstarg, secondarg)


Test()
```
This code works, but pyright reports that test() has only one expected arg instead of 2.

It probably thinks it needs to strip away "self"? but as its a staticmethod this is not the case.

Or maybe i typed the decorator wrong</div>

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.207, which I just published. It will also be included in the next release of pylance.
